internal/os/disk/api.go

```diff
@@ -254,6 +254,7 @@ func (imp APIImplementor) GetDiskNumberWithID(page83ID string) (uint32, error) {
 
 	for i := range disks {
 		h, err := syscall.Open(disks[i].Path, syscall.O_RDONLY, 0)
+		defer syscall.Close(h)
```

Comment:
What's the maximum number of disks a node can have? Would it be worth enclosing this loop in a code section so that it gets closed at the end of each loop?

Reply:
for ec2 depends on instance type but typically like 20-30
https://docs.aws.amazon.com/AWSEC2/latest/WindowsGuide/volume_limits.html#instance-type-volume-limits (EKS only supports windows nitro instances so the "AWS PV" driver in the other section don't apply)

do you mean to wrap it like this? I don't think it's worth the loss in readability 
```
func() (uint32, error) {
   ...
   syscall.Open
   defer syscall.Close(h)
}()
```

Reply:
Or a separate function could also work.

The default fd limit per process is 1024 for Linux. I don't know about Windows. Does Windows also have a maximum disk limit? cc @ddebroy @jingxu97

Reply:
here it says that it's 16M https://docs.microsoft.com/en-us/archive/blogs/markrussinovich/pushing-the-limits-of-windows-handles#maximum-number-of-handles

Reply:
did some refactoring so we always open and close 1 handle at a time